File: benchmarks/tsb/bench_period_index_range.ts
/**
 * Benchmark: PeriodIndex.periodRange / PeriodIndex.fromPeriods — PeriodIndex construction.
 * Outputs JSON: {"function": "period_index_range", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Period, PeriodIndex } from "../../src/index.ts";

const WARMUP = 5;
const ITERATIONS = 50;

const startPeriod = Period.fromDate(new Date(Date.UTC(2000, 0, 1)), "D");
const startMonth = Period.fromDate(new Date(Date.UTC(2000, 0, 1)), "M");
const dayPeriods = Array.from({ length: 365 * 10 }, (_, i) =>
  Period.fromDate(new Date(Date.UTC(2000, 0, 1) + i * 86_400_000), "D"),
);

for (let i = 0; i < WARMUP; i++) {
  PeriodIndex.periodRange(startPeriod, 3650);
  PeriodIndex.periodRange(startMonth, 120);
  PeriodIndex.fromPeriods(dayPeriods.slice(0, 365));
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  PeriodIndex.periodRange(startPeriod, 3650);
  PeriodIndex.periodRange(startMonth, 120);
  PeriodIndex.fromPeriods(dayPeriods.slice(0, 365));
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "period_index_range",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_period_index_range.py
"""Benchmark: pd.period_range / pd.PeriodIndex — PeriodIndex construction."""
import json
import time
import pandas as pd

WARMUP = 5
ITERATIONS = 50

start_d = pd.Period("2000-01-01", freq="D")
start_m = pd.Period("2000-01", freq="M")
day_periods = pd.period_range(start="2000-01-01", periods=365 * 10, freq="D")

for _ in range(WARMUP):
    pd.period_range(start=start_d, periods=3650, freq="D")
    pd.period_range(start=start_m, periods=120, freq="ME")
    pd.PeriodIndex(day_periods[:365])

start = time.perf_counter()
for _ in range(ITERATIONS):
    pd.period_range(start=start_d, periods=3650, freq="D")
    pd.period_range(start=start_m, periods=120, freq="ME")
    pd.PeriodIndex(day_periods[:365])
total = (time.perf_counter() - start) * 1000

print(json.dumps({"function": "period_index_range", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))
